src/shareManagement.js

```diff
@@ -90,18 +90,26 @@ class Shares {
    */
   shareFileWithUser (path, username, optionalParams) {
     path = this.helpers._normalizePath(path)
+    let urlPath = 'shares'
 
     let postData = {
       shareType: this.helpers.OCS_SHARE_TYPE_USER,
-      shareWith: username,
-      path: path
+      shareWith: username
+    }
+
+    if (!optionalParams || !optionalParams.spaceRef) {
+      postData.path = path
     }
 
     if (optionalParams) {
+      if (optionalParams.spaceRef) {
+        urlPath += `?space_ref=${optionalParams.spaceRef}`
+      }
+
       postData = { ...postData, ...this._getOptionalParams(optionalParams) }
     }
 
```

Comment:
a combination of Lodash _.snakeCase() and  https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams might be more programmatic

Reply:
I like the idea with the `URLSearchParams` and implemented those 👍 
However, I prefer keeping `space_ref` hardcoded because `spaceRef` is basically just a variable naming, so I don't like those to be dependent on each other.